app/src/main/java/fr/free/nrw/commons/nearby/presenter/NearbyParentFragmentPresenter.java

```diff
@@ -373,4 +373,12 @@ public boolean areLocationsClose(LatLng cameraTarget, LatLng lastKnownLocation)
             return true;
         }
     }
+
+    public boolean isNearbyLocked() {
```

Comment:
Exposing code for test is a code smell. The option is to rewrite or use the nasty `@VisibleForTesting` annotation.

I'd say rewrite because `lockUnlockNearby` sets it quite easily for you.

Reply:
I didn't know about the annotation will do! Thanks:)

Reply:
I meant for you to rewrite the test, `@VisibleForTesting` is an ugly bandaid and should only be used in extreme scenarios